### PR TITLE
enhance: support null expr

### PR DESCRIPTION
**Pull request by smellthemoon**
#31728

### CODE REVIEW

**Comment by smellthemoon** [comment]
rerun go-sdk

File: internal/core/src/exec/expression/NullExpr.cpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    if (expr_->op_ == proto::plan::NullExpr_NullOp_IsNull) {
+        res.flip();
+    }
+    auto res_vec =
+        std::make_shared<ColumnVector>(std::move(res), std::move(valid_res));
+    return res_vec;
+}
+
+ColumnVectorPtr
+PhyNullExpr::PreCheckNullable() {
```
Comment (tedxu) [inline internal/core/src/exec/expression/NullExpr.cpp:101]: What does this function do?

File: internal/core/unittest/test_expr.cpp [hunk 2]
```diff
@@ -5138,6 +5138,215 @@ TEST_P(ExprTest, TestBinaryArithOpEvalRangeBenchExpr) {
     }
 }
 
+TEST(Expr, TestExprNull) {
+    auto schema = std::make_shared<Schema>();
+    auto int8_fid = schema->AddDebugField("int8", DataType::INT8, true);
```
Comment (tedxu) [inline internal/core/unittest/test_expr.cpp:5143]: We should add bool type related tests here.

File: internal/parser/planparserv2/Plan.g4 [hunk 3]
```diff
@@ -74,6 +76,9 @@ BXOR: '^';
 AND: '&&' | 'and';
 OR: '||' | 'or';
 
+ISNULL: 'is_null' | 'IS_NULL';
```
Comment (tedxu) [inline internal/parser/planparserv2/Plan.g4:79]: Do we need to support grammar like `col IS NULL`, instead of `IS_NULL(col)`?

File: internal/core/src/exec/expression/NullExpr.cpp (hunk 1 above)
Comment (smellthemoon) [reply internal/core/src/exec/expression/NullExpr.cpp:101]: To check if nullable is false, then the TargetBitmap is all true when is_not_null, and all false when is_null.

File: internal/parser/planparserv2/Plan.g4 (hunk 3 above)
Comment (smellthemoon) [reply internal/parser/planparserv2/Plan.g4:79]: it's hard to parse col name if col is ahead of ISNULL

Comment (smellthemoon) [reply internal/parser/planparserv2/Plan.g4:79]: and I think IS_NULL(col) format is enough for users.

**Review by tedxu — COMMENTED** [review]
/lgtm
